### PR TITLE
Dimitrios/test teacher GET endpoints

### PR DESCRIPTION
**Pull request by dimitris4**

### CODE REVIEW

File: tests/users-endpoints.test.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                    reject(error);
+                }
+                resolve(result.insertId);
+            });
+            db.release();
+        });
+    })
+}
+
+const saveLectureToDB = (teacherId, dateTime) => {
```
Comment (dprzygocka) [inline tests/users-endpoints.test.js]: this method fails bc you don't have any courses inserted in database, pls fix it by adding them bedore you add the leactures

File: tests/users-endpoints.test.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
+const server = require("../app");
+const supertest = require("supertest");
+
+const {pool} = require('../database/connection');
+
+const saveTeacherToDB = () => {
+    return new Promise((resolve, reject) => {
+        pool.getConnection((err, db) => {
+            let query =
+                'INSERT INTO heroku_f94932c1d70fb93.users (first_name, last_name, email, user_role, password, class_id) ' +
```
Comment (dprzygocka) [inline tests/users-endpoints.test.js]: remove heroku_f94932c1d70fb93 name from query bc we will use different db

File: tests/users-endpoints.test.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+const saveLectureToDB = (teacherId, dateTime) => {
+    let lecture = {
+        lecture_id: undefined,
+        start_date_time: dateTime,
+        course_id: 1,
+        class_id: 1,
+    }
+    return new Promise((resolve, reject) => {
+        pool.getConnection((err, db) => {
+            let query = `INSERT INTO heroku_f94932c1d70fb93.lectures (teacher_id, start_date_time, course_id, class_id) VALUES (${teacherId},"${dateTime}", 1, 1);`;
```
Comment (dprzygocka) [inline tests/users-endpoints.test.js]: remove heroku_f94932c1d70fb93 name from query bc we will use different db

File: tests/users-endpoints.test.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            });
+            db.release();
+        });
+    })
+}
+
+const deleteLectureFromDB = (teacherId) => {
+    return new Promise((resolve, reject) => {
+        pool.getConnection((err, db) => {
+            let query = `DELETE FROM heroku_f94932c1d70fb93.lectures WHERE teacher_id = ${teacherId};`;
```
Comment (dprzygocka) [inline tests/users-endpoints.test.js]: remove heroku_f94932c1d70fb93 name from query bc we will use different db and don't remove only the teacher you added, remove them all

File: tests/users-endpoints.test.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+            });
+            db.release();
+        });
+    })
+}
+
+const deleteTeacherFromDB = (teacherId) => {
+    return new Promise((resolve, reject) => {
+        pool.getConnection((err, db) => {
+            let query = `DELETE FROM heroku_f94932c1d70fb93.users WHERE user_id = ${teacherId};`;
```
Comment (dprzygocka) [inline tests/users-endpoints.test.js]: remove heroku_f94932c1d70fb93 name from query bc we will use different db and don't remove only the teacher you added, remove them all

File: tests/users-endpoints.test.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    const teacherId = await saveTeacherToDB();
+    const lecture = await saveLectureToDB(teacherId, dateTime);
+    await supertest(server).get(`/api/users/lectures/${teacherId}`)
+        .expect(200)
+        .then((response) => {
+            expect(Array.isArray(response.body)).toBeTruthy();
+            expect(response.body.lecture_id).toEqual(lecture.lecture_id);
+            expect(response.body.start_date_time).toEqual(dateTime);
+            expect(response.body.name).toEqual('Development of Large Systems');
+        }).catch(async () => {
```
Comment (dprzygocka) [inline tests/users-endpoints.test.js]: don't need to delete from db here, pls just throw error stack: e.stack()

File: tests/users-endpoints.test.js [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+                if (!error) {
+                    resolve(true);
+                }
+            });
+            db.release();
+        });
+    })
+}
+
+test("GET /api/users/lectures/:teacherId", async () => {
```
Comment (dprzygocka) [inline tests/users-endpoints.test.js]: pls add more test cases: test what will happen if if the date is not today, test with array where we have multiple lecutes for today, test teacher id if it's string or null,
if array is empty, combination of classes today and not today

Comment (dprzygocka) [inline tests/users-endpoints.test.js]: put all test into describe so we can close the connection after the tests run, pls check integration branch for examples (teacherStatistics or login or register file)

File: tests/users-endpoints.test.js [hunk 8]
```diff
@@ -0,0 +1,119 @@
+const server = require("../app");
```
Comment (dprzygocka) [inline tests/users-endpoints.test.js]: pls split the endpoints to different files with relevant name, and use extension for the files name.spec.js

File: tests/users-endpoints.test.js [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+    const teacherId = await saveTeacherToDB();
+    const lecture = await saveLectureToDB(teacherId, dateTime);
+    await supertest(server).get(`/api/users/classes/courses/all/${teacherId}`)
+        .expect(200)
+        .then((response) => {
+            expect(Array.isArray(response.body)).toBeTruthy();
+            expect(response.body.course_id).toEqual(1);
+            expect(response.body.class_id).toEqual(1);
+            expect(response.body.courseName).toEqual('Development of Large Systems');
+        }).catch(async () => {
```
Comment (dprzygocka) [inline tests/users-endpoints.test.js]: don't need to delete from db here, pls just throw error stack: e.stack()

File: tests/users-endpoints.test.js [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+            expect(response.body.name).toEqual('Development of Large Systems');
+        }).catch(async () => {
+            await deleteLectureFromDB(teacherId);
+            await deleteTeacherFromDB(teacherId);
+        });
+    await deleteLectureFromDB(teacherId);
+    await deleteTeacherFromDB(teacherId);
+}, 20000);
+
+test("GET /api/users/classes/courses/all/:teacherId", async () => {
```
Comment (dprzygocka) [inline tests/users-endpoints.test.js]: pls add more test cases: test what will happen if there are more combinations of data in db, , multiple lectures for the same class - course combination and for different combinations, test teacher id if it's string or null,
if array is empty,

File: tests/users-endpoints.test.js [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+    const dateTime = getDateTime();
+    const teacherId = await saveTeacherToDB();
+    const lecture = await saveLectureToDB(teacherId, dateTime);
```
Comment (dprzygocka) [inline tests/users-endpoints.test.js]: add courses in db

File: tests/users-endpoints.test.js [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+        }).catch(async () => {
+            await deleteLectureFromDB(teacherId);
+            await deleteTeacherFromDB(teacherId);
+        });
+    await deleteLectureFromDB(teacherId);
+    await deleteTeacherFromDB(teacherId);
+}, 20000);
+
+// return a string representing a datetime on the same day
+const getDateTime = () => {
```
Comment (dprzygocka) [inline tests/users-endpoints.test.js]: move this declaration, the definitation should be before the usage

File: app.js [hunk 13]
```diff
@@ -9,10 +9,6 @@ app.use(cors({origin: "*"})); // https://www.youtube.com/watch?v=PNtFSVU-YTI
 const { Utils } = require('./models/Utils');
```
Comment (dprzygocka) [inline app.js:9]: IMPORTANT! PLS DO NOT USE THE CONNECTION TO REAL DB, create new test db locally or connect to azure (the one created by gabriel), run the script and change env variables for the connection to test the api

**Review by dprzygocka — CHANGES_REQUESTED** [review]

File: tests/users-endpoints.test.js (hunk 8 above)
Comment (dimitris4) [reply tests/users-endpoints.test.js]: why not name.test.js as it is now?

**Pull request merged**